staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/validation.go

```diff
@@ -85,8 +86,10 @@ func validate(pth *field.Path, s *structuralschema.Structural, rootSchema *struc
 				allErrs = append(allErrs, field.Invalid(pth.Child("default"), s.Default.Object, fmt.Sprintf("must result in valid metadata: %v", errs.ToAggregate())))
 			} else if errs := apiservervalidation.ValidateCustomResource(pth.Child("default"), s.Default.Object, validator); len(errs) > 0 {
 				allErrs = append(allErrs, errs...)
-			} else if celValidator := cel.NewValidator(s); celValidator != nil {
-				allErrs = append(allErrs, celValidator.Validate(pth.Child("default"), s, s.Default.Object)...)
+			} else if celValidator := cel.NewValidator(s, cel.PerCallLimit); celValidator != nil {
+				err, rmCost := celValidator.Validate(pth.Child("default"), s, s.Default.Object, remainingCost)
+				allErrs = append(allErrs, err...)
+				remainingCost -= rmCost
```

Comment:
is `-=` here correct? I thought this should be `remainingCost = rmCost`, and short-circuiting if < 0

if this is currently wrong, need a test exercising this

Reply:
I have fixed it. And will add test in https://github.com/kubernetes/kubernetes/pull/108646 when able to pass context.

Reply:
It is a little tricky to add test here since it is hard to pass cost limit from crd validation and shouldn't. Do we wanna make cost limit as an input arg here for testing purpose?

Reply:
Test added.